lib/widgets/message_list.dart

```diff
@@ -24,10 +24,28 @@ class MessageListPage extends StatelessWidget {
 
   final Narrow narrow;
 
+  @override
+  State<MessageListPage> createState() => MessageListPageState();
+}
+
+class MessageListPageState extends State<MessageListPage> {
+  GlobalKey<ComposeBoxController> get composeBoxKey => _composeBoxKey;
+  final GlobalKey<ComposeBoxController> _composeBoxKey = GlobalKey();
+
+  /// A [ComposeBoxController], if this [MessageListPage] offers a compose box.
+  ///
+  /// Uses the inefficient [BuildContext.findAncestorStateOfType];
+  /// don't call this in a build method.
+  static ComposeBoxController? composeBoxControllerOf(BuildContext context) {
+    final messageListPageState = context.findAncestorStateOfType<MessageListPageState>();
+    assert(messageListPageState != null, 'No MessageListPageState ancestor');
```

Comment:
Oh, related to leaving the state private:
```suggestion
    assert(messageListPageState != null, 'No MessageListPage ancestor');
```
the error message can just stick to the public name, of the widget.

(See for comparison `PerAccountStoreWidget.of`.)